tsl/src/nodes/decompress_chunk/compressed_batch.c

```diff
@@ -224,6 +275,50 @@ decompress_column(DecompressContext *dcontext, DecompressBatchState *batch_state
 	}
 }
 
+/*
+ * When we have a dictionary-encoded Arrow Array, and have run a predicate on
+ * the dictionary, this function is used to translate the dictionary predicate
+ * result to the final predicate result.
+ */
+static void
+translate_bitmap_from_dictionary(const ArrowArray *arrow, uint64 *restrict dict_result,
+								 uint64 *restrict final_result)
+{
+	Assert(arrow->dictionary != NULL);
+
+	const size_t n = arrow->length;
+	int16 *restrict indices = (int16 *) arrow->buffers[1];
+	for (size_t outer = 0; outer < n / 64; outer++)
+	{
+		uint64 word = 0;
+		for (size_t inner = 0; inner < 64; inner++)
+		{
+			const size_t row = outer * 64 + inner;
+			const size_t bit_index = inner;
+#define INNER_LOOP                                                                                 \
```

Comment:
Could you add a comment on why this macro is needed and why an inlined function cannot be used?

Reply:
It's going to be a function with 5 parameters and one of them in/out, so to my taste a macro is more readable in this case.  Added a comment about this.